src/handlers/records-query.ts

```diff
@@ -38,6 +39,11 @@ export class RecordsQueryHandler implements MethodHandler {
       // authentication
       try {
         await authenticate(message.authorization!, this.didResolver);
+
+        // Only run protocol authz if message deliberately invokes it
+        if (RecordsQueryHandler.shouldProtocolAuthorizeQuery(recordsQuery)) {
+          await ProtocolAuthorization.authorizeQuery(tenant, recordsQuery, this.messageStore);
```

Comment:
To preserve existing functionality, we only run authz and throw errors if the message invokes a `protocolRole`. If the message does not invoke a `protocolRole`, we simply skip querying of protocol records.

Reply:
> we simply skip querying of protocol records.

Sanity: Not sure if I understand what this means exactly. You don't mean we "explicitly exclude" protocol records if message doesn't invoke `protocolRole` right?

Reply:
We don't exclude protocol records when `protocolRole` is absent. It's that `protocolRole` must be present for the request to get records beyond published records, records they authored, and records they received.